app/post.go

```diff
@@ -359,6 +359,9 @@ func (a *App) CreatePost(post *model.Post, channel *model.Channel, triggerWebhoo
 		a.SendEphemeralPost(post.UserId, ephemeralPost)
 	}
 
+	// Notify shared channel sync service for the new change.
+	a.NotifySharedChannelSync(channel, model.WEBSOCKET_EVENT_POSTED)
```

Comment:
We need to explicitly invoke a sync here, in order to cover the case were, the Leader is the one that handled the user intention(i.e create a post)

Reply:
I don't believe this is actually needed. This will actually generate duplicate notifications. You would only need to call `ServerSyncSharedChannelHandler` at a single place - `PublishSkipClusterSend`. All events are routed through this method. There's no need to call `NotifySharedChannelSync` here or anywhere else.

In short, this is how the cluster publishing happens:
- Everything starts off with the `(a *App) Publish` call.
- Now this call splits into two - `s.PublishSkipClusterSend` and `s.Cluster.SendClusterMessage`.
- The `PublishSkipClusterSend` handles propagating of events for the local machine itself. And `s.Cluster.SendClusterMessage` sends it off to the cluster.
- Now in the cluster handler, everything is routed through `PublishSkipClusterSend` again.
- So therefore, _all_ events, irrespective of being generated locally, or coming from another node, will be routed through `PublishSkipClusterSend`.
- So if we call `a.ServerSyncSharedChannelHandler(event)` inside `PublishSkipClusterSend`, we will automatically get free handling of all events. All we need to do is filter the right event, and check if it's a leader or not, which we are already doing.